### ISSUE TITLE
Web frontend 

### ISSUE THREAD

**Issue by patrickpr** [J-8/ShoppingList_backend#69]
Hi,

The mobile app is great but it's a real pain to input a full list on a phone : is there some plan to make a web version (to make the input with a real keyboard) ?
Didn't see anything like this on the roadmap.

**Comment by jklmnn** [comment]
That would be possible but we currently have many to do for the next version and only few time, so there is currently no plan.

**Comment by J-8** [comment]
Hey,  thanks for your idea. Jklmnn is right, there is a lot to do and fix which is more important right now. But I see no harm in thinking about that at a later date.
And you are always welcome to come up with something!

**Comment by ghost** [comment]
I think good idead, but for this the must a lot of steps before. You must also great a lot of SQL Updates like an user table for login reasons and stuff like this.
Maybe in later versions.

**Comment by jklmnn** [comment]
An easier way to provide use on a desktop would be a desktop app. That could be done independently from the server. Still someone must do is. I started to create one once, but I didn't come far.

**Comment by ghost** [comment]
In Java?

**Comment by ghost** [comment]
or python

**Comment by patrickpr** [comment]
Ok, I'll work on this. Here is my idea : 
- Simple web page using jqueryUI / PHP
- Authentication using authkey like the mobile app
- Entry page (index.php ?) at same level as api.php and javascript/php lib in /lib/web/ (?)

Comments are welcome !

**Comment by ghost** [comment]
I think a better way is, that the websites create qrcode and app will create with that code a one time password to login.

**Comment by J-8** [comment]
Isn't that unnecessarily complicated? So you would need a one time password every time you want to add something in the browser? If the regular password is used, there is no need for users or anything.
Actually, do whatever you think is best. Then we have something to start with :)

**Comment by patrickpr** [comment]
Hi, 
Starting the interface (it IS ugly and it's not the point) : 

![capture](https://cloud.githubusercontent.com/assets/2913330/10546239/21bd265e-742d-11e5-8f58-b4bde309be19.PNG)

Login is made with the auth key (no user - not shown)
All items are ordered by category, category change with drag&drop.
All actions are automatically saved (no commit)
Uses api.php

Some questions : 
- I assume the categories (and "checked") will be operational (as defined in DB connector), is there a planning for the api.php release with category ?
- To use api.php with ajax, I must store the authkey in clear on the web page. I don't like that. Is there a plan to allow transmit of digest only ? If not I can use session to bypass authentication in api.php.
- I think it would be really useful to add units (liter, kg etc...) along with quantity (it's totally unrelated).

Comments are welcome !

**Comment by J-8** [comment]
Great work!
Category is optional, if it is not set, the app will just categorize it as such. The checked Boolean is what is used to indicate if the item has been crossed of the list, but not deleted.
The thing that is missing is the api.php that supports categories. I haven't had the time to change that.

I have thought about the units as well. But I am not sure if that would not make the interface too cluttered... I would have to play around or change the current input completely. Maybe leave the list and use a completely new window/activity.

**Comment by J-8** [comment]
Great work!
Category is optional, if it is not set, the app will just categorize it as such. The checked Boolean is what is used to indicate if the item has been crossed of the list, but not deleted.
The thing that is missing is the api.php that supports categories. I haven't had the time to change that.

I have thought about the units as well. But I am not sure if that would not make the interface too cluttered... I would have to play around or change the current input completely. Maybe leave the list and use a completely new window/activity.

**Comment by patrickpr** [comment]
I didn't manage to check the password with the api.php code (but didn't try very long !). Anyway, I changed api.php so the password is not sent at every request. This can be changed very quickly if needed.

I pushed the first version for review.

Units are really useful I think (3 apples or 3 Kg of apples ? )

Also : it could be possible to use the web frontend to change password, display mobile appli Qrcode again, etc...
